### PR TITLE
Implement create approval voting

### PR DESCRIPTION
**Pull request by selankon**
Fix https://github.com/vocdoni/ui-scaffold/issues/595

Implements approval voting on process creation form.

- Refactor approval voting as implemented feature
- Implements components for SingleChoice or ApprovalVoting on questions page
- It implements refactors in order to share components structure between pages and question types

It refactors a bit to reuse the code from Census type form and create generic components to use also on Questions pages. This is because both had same structure (a tab screen with a list of  implemented an unimplemented features).

### Refactor of question types

The old `index.tsx` that implements the logic for the single choice questions is moved inside the `Question.tsx` file, creating a `QuestionsPage` component shared between the questions types. 

So using this new component it implements two new questions types:

- [SingleChoice](https://github.com/vocdoni/ui-scaffold/pull/693/files#diff-3a18c6c306af6a1a4fd2c48b3af65f5663615561b87d8319cef0a0101f2d1667)
- [ApprovalVoting](https://github.com/vocdoni/ui-scaffold/pull/693/files#diff-28f6340daeed1557779ddc87c6dd78d63fd4088f5a167dda63d343e079bac3bd)

### What is refactored from Census form?

Using as example the [Census.tsx](L59-L155) component, I created the more generic [TabsPage](https://github.com/vocdoni/ui-scaffold/blob/7927f33e7cb88948c417863f31ddf60f90269a17/src/components/ProcessCreate/Steps/TabsPage.tsx) component which is basically an skeleton for a page that have:

- Title and description
- A list of implemented feature tabs
- A list of unimplemented features tabs that open a modal

This is how it looks like

https://github.com/vocdoni/ui-scaffold/blob/7927f33e7cb88948c417863f31ddf60f90269a17/src/components/ProcessCreate/Steps/Census.tsx#L12-L39

This component is used on the Questions step and on the Census step.

### CODE REVIEW

**Review by jpaulet — CHANGES_REQUESTED** [review]
Missing translations and we should select, by default, the "Single choice" voting, instead of none:

image

File: src/components/ProcessCreate/Questions/Question.tsx [hunk 1]
```diff
@@ -74,4 +77,86 @@ const Question = ({ index, remove }: Props) => {
...
+  }, [questions, append, tabIndex])
+
+  useEffect(() => {
+    const questionErrors = errors.questions as FieldErrors<CustomFieldError>[] | undefined
+
+    if (!questionErrors) return
+
+    const firstError = questionErrors.findIndex((curr) => typeof curr !== 'undefined')
+    setTabIndex(firstError)
+  }, [errors.questions])
```
Comment (elboletaire) [inline src/components/ProcessCreate/Questions/Question.tsx:125]: Friendly reminder to always comment effects :)

Comment (selankon) [reply src/components/ProcessCreate/Questions/Question.tsx:125]: Afair, I copied this effects from somewhere but i commented them anyway. Thanks for the reminder

**Comment by selankon** [comment]
> Missing translations and we should select, by default, the "Single choice" voting, instead of none:

@jpaulet added translations and set default to single choice

**Review by jpaulet — APPROVED** [review]

**Comment by elboletaire** [comment]
This is still shown without proper interpolation of html tags
imatge
imatge

**Pull request merged**